daemon/graphdriver/counter.go

```diff
@@ -1,32 +1,69 @@
 package graphdriver
 
-import "sync"
+import (
+	"sync"
+
+	"github.com/docker/docker/pkg/mount"
+)
+
+type minfo struct {
+	check bool
+	count int
+}
 
 // RefCounter is a generic counter for use by graphdriver Get/Put calls
 type RefCounter struct {
-	counts map[string]int
+	counts map[string]*minfo
 	mu     sync.Mutex
 }
 
 // NewRefCounter returns a new RefCounter
 func NewRefCounter() *RefCounter {
-	return &RefCounter{counts: make(map[string]int)}
+	return &RefCounter{counts: make(map[string]*minfo)}
 }
 
 // Increment increaes the ref count for the given id and returns the current count
-func (c *RefCounter) Increment(id string) int {
+func (c *RefCounter) Increment(path string) int {
 	c.mu.Lock()
-	c.counts[id]++
-	count := c.counts[id]
+	m := c.counts[path]
+	if m == nil {
+		m = &minfo{}
+		c.counts[path] = m
+	}
+	// if we are checking this path for the first time check to make sure
+	// if it was already mounted on the system and make sure we have a correct ref
+	// count if it is mounted as it is in use.
+	if !m.check {
+		m.check = true
+		mntd, _ := mount.Mounted(path)
```

Comment:
sure, but there are many reasons why this is worth the cost for getting the correct results so i don't know what to tell ya